internal/pkg/levee/propagation/propagation.go

```diff
@@ -243,6 +246,45 @@ func (prop *Propagation) visitOperands(n ssa.Node, maxInstrReached map[*ssa.Basi
 	}
 }
 
+func (prop *Propagation) visitSelect(sel *ssa.Select, maxInstrReached map[*ssa.BasicBlock]int, lastBlockVisited *ssa.BasicBlock) {
+	// The Send operations have to be processed before the Recv operations in case a channel
+	// is written to in one branch and read from in a different branch. This matters if the
+	// branches are reachable from each other, e.g. when the select is in a loop.
```

Comment:
This behavior is covered by existing tests, see `TestTaintedAndSinkedInDifferentBranches(InLoop)`.